plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js

```diff
@@ -1591,6 +1591,89 @@ jQuery( function ( $ ) {
 					return settings;
 				}
 			});
+
+			$( '#order_custom #metakeyselect').filter( ':not(.enhanced)' ).each( function() {
+				var select2Element = this;
+				var select2_args = {
+					allowClear: !!$(this).data('allow_clear'),
+					placeholder: $(this).data('placeholder'),
+					escapeMarkup: function (m) {
+						return m;
+					},
+					ajax: {
+						url: wc_enhanced_select_params.ajax_url,
+						dataType: 'json',
+						delay: 500,
+						data: function (params) {
+							return {
+								order_id: $(this).data('order_id'),
+								action: 'woocommerce_json_search_order_metakeys',
+								security: wc_enhanced_select_params.search_order_metakeys_nonce
+							};
+						},
+						success: function (data) {
+							var terms = [];
+							if (data) {
+								$.each(data, function (id, term) {
+									terms.push({
+										id: term,
+										text: term,
+									});
+								});
+							}
+							// Reinitialize with the loaded data to avoid continued ajax requests when searching since
+							// we are not using the search term to filter the list on the backend.
+							select2_args.data = terms;
+							delete select2_args.ajax;
+							$(select2Element).selectWoo(select2_args).select2('open');
+						},
+						cache: true
+					},
+					language: {
+						errorLoading: function () {
+							// Workaround for https://github.com/select2/select2/issues/4355 instead of i18n_ajax_error.
+							return wc_enhanced_select_params.i18n_searching;
+						},
+						inputTooLong: function (args) {
```

Comment:
I wonder with how often these get repeated if there's a way we can reuse some default settings across Select2 instances.

Does not need to be a blocker for this PR, just thinking out loud.